facet-core/src/types/mod.rs

```diff
@@ -119,6 +119,9 @@ pub enum ShapeAttribute {
     /// missing in the input should be filled with corresponding field values from
     /// a `T::default()` (where T is this shape)
     Default,
+    /// Indicates that, this shape should be serialized with its `ToString`
+    /// implementation, and deserialized with its `FromStr` implementation
+    AsString,
```

Comment:
I don't love that this is another `ShapeAttribute` instead of a `ShapeFlag` but I guess that refactor hasn't happened yet.